tests/queries/tests.py

```diff
@@ -2398,6 +2398,12 @@ def test_named_values_list_expression(self):
         values = qs.first()
         self.assertEqual(values._fields, ('combinedexpression2', 'combinedexpression1'))
 
+    def test_named_values_pickle_ticket_31843(self):
+        qs = Number.objects.extra(select={'num3': 'num+3'})
```

Comment:
There is no need to mix these tests with `.extra()` we can use existing columns, e.g.
```python
values = Number.objects.values_list('num', 'other_num', named=True).get()
```